bindings/rust/s2n-tls/src/session_ticket.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::ptr::NonNull;
+use std::time::Duration;
+
+use s2n_tls_sys::*;
+
+use crate::{
+    connection::Connection,
+    error::{Error, Fallible},
+};
+
+/// A trait to retrieve session tickets from the connection
+pub trait SessionTicketCallback {
+    fn on_session_ticket(&self, connection: &mut Connection, session_ticket: SessionTicket);
+}
+
+/// Wrapper around a session ticket struct
+#[derive(Debug)]
+pub struct SessionTicket {
```

Comment:
What's the purpose of this struct? There's currently no way to get owned data out so we end up copying in quite a few places here. Also: would it be better for the application to own the allocation since they may want to use something like `Bytes` instead?